frontend/src/tests/instructor-permission-test.js

```diff
@@ -600,37 +600,36 @@ export function instructorsPermissionTests(api) {
             );
 
             // Update the DDAH and check it was updated correctly
+            const updatedDuties = [
+                {
+                    order: 1,
+                    hours: 40,
+                    description: "marking:Test Marking",
+                },
+                {
+                    order: 2,
+                    hours: 30,
+                    description: "other:Additional duties",
+                },
+            ];
             const ddahWithRestrictedFields = {
                 assignment_id: assignment.id,
-                duties: [
-                    {
-                        order: 1,
-                        hours: 40,
-                        description: "marking:Test Marking",
-                    },
-                    {
-                        order: 2,
-                        hours: 30,
-                        description: "other:Additional duties",
-                    },
-                ],
+                duties: updatedDuties,
                 approved_date: new Date().toISOString(),
                 accepted_date: "2020-09-02T00:00:00.000Z",
                 revised_date: "2020-09-01T00:00:00.000Z",
                 emailed_date: "2020-08-28T00:00:00.000Z",
                 signature: "Harry Potter",
             };
             resp = await apiPOST(`/instructor/ddahs`, ddahWithRestrictedFields);
-            expect(resp).toHaveStatus("success");
-            expect(resp.payload).toEqual(expect.objectContaining(originalDdah));
 
-            resp = await apiGET(`/instructor/sessions/${session.id}/ddahs`);
-            // still a success, but does not change the underlying data
+            // We are looking for a success, but expect only duties to change
             expect(resp).toHaveStatus("success");
-            const newDdah = resp.payload.find(
-                (ddah) => ddah.assignment_id === assignment.id
-            );
-            expect(newDdah).toEqual(expect.objectContaining(originalDdah));
+            const originalDdahWithUpdatedDuties = {
+                ...originalDdah,
+                duties: updatedDuties,
+            };
+            expect(resp.payload).toEqual(originalDdahWithUpdatedDuties);
```

Comment:
Here we are doing pretty much the same test as before, but also checking that duties updates are still permitted whereas other updates (dates and signature) are not.